hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -341,10 +343,224 @@ public IdentifierHelper buildIdentifierHelper(
 		builder.applyIdentifierCasing( dbMetaData );
 
 		builder.applyReservedWords( dbMetaData );
+		builder.applyReservedWords( getDerbyKeywords() );
 		builder.applyReservedWords( getKeywords() );
 
 		builder.setNameQualifierSupport( getNameQualifierSupport() );
 
 		return builder.build();
 	}
+
+	protected Set<String> getDerbyKeywords() {
+		final Set<String> derbyKeyword = new HashSet<String>(  );
```

Comment:
Why a new "group" of keywords rather than just adding to the already existing keywords set?